firmware/Sources/HAL/HAL_PWM/hal_pwm.c

```diff
@@ -0,0 +1,104 @@
+/*                              Include other headers                             */
+/**********************************************************************************/
+#include <stdint.h>
+
+/**********************************************************************************/
+/*                     Definition of local symbolic constants                     */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Definition of local function like macros                    */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*            Definition of local types (typedef, enum, struct, union)            */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                         Definition of local variables                          */
+/**********************************************************************************/
+static HRTIM_CompareCfgTypeDef pCompareCfg = {0};
+/**********************************************************************************/
+/*                        Definition of exported variables                        */
+/**********************************************************************************/
+extern HRTIM_HandleTypeDef hhrtim1;
+
+/**********************************************************************************/
+/*                      Definition of exported constant data                      */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Declaration of local function prototypes                    */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                       Definition of local constant data                        */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                         Definition of local functions                          */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                        Definition of exported functions                        */
+/**********************************************************************************/
+
+HAL_PWM_result_e HAL_PwmInit(void){
+	HAL_PWM_result_e res = HAL_PWM_RESULT_SUCCESS;
```

Comment:
change logict to init with HAL_PWM_RESULT_ERROR

Reply:
Done